ansible/files/bootstrap/telemetry.sh

```diff
@@ -94,13 +158,20 @@ pyliblzma==0.5.3
 plotly==1.6.16
 seaborn==0.6.0
 EOF
-$ANACONDAPATH/bin/pip install -r $PIP_REQUIREMENTS_FILE
-rm $ANACONDA_SCRIPT
+
+$ANACONDA_PATH/bin/pip install -r $PIP_REQUIREMENTS_FILE
+rm /mnt/$ANACONDA_SCRIPT
 rm $PIP_REQUIREMENTS_FILE
 
-# Add public key
-if [ -n "$PUBLIC_KEY" ]; then
-    echo $PUBLIC_KEY >> $HOME/.ssh/authorized_keys
+AUTH_KEYS_PATH="$HOME/.ssh/authorized_keys"
+
+# Add public key if it's not currently there
+if [ -n "$PUBLIC_KEY" ] && ! sudo grep -q "$PUBLIC_KEY" "$AUTH_KEYS_PATH" ; then
+    sudo echo $PUBLIC_KEY | sudo tee -a "$AUTH_KEYS_PATH"
```

Comment:
I don't think we need the `sudo echo` here. `echo` alone should be able output the env variable.